mirage/serializers/application.ts

```diff
@@ -21,15 +21,15 @@ export default class ApplicationSerializer extends JSONAPISerializer {
     }
 
     buildRelatedLinkMeta<T extends ModelRegistry[keyof ModelRegistry]>(
-        model: T,
+        model: ModelInstance<T>,
         relationship: RelationshipsFor<T>,
     ): RelatedLinkMeta {
         let relatedCounts: string[] = [];
         if (this.request.queryParams.related_counts) {
             relatedCounts = this.request.queryParams.related_counts.split(',');
         }
-        const related = model[relationship];
-        const count = Array.isArray(related) ? related.length : 0;
+        const related = (model as Model<T>)[relationship];
```

Comment:
I'm confused... is this a `Model` or a `ModelInstance`? Should `ModelInstance` extend `Model`?

Reply:
A `ModelInstance` *is* a `Model`, but here we have to disambiguate because TS doesn't know that `relationship` isn't something defined in `ModelInstanceShared` that doesn't have a `models` property.